### PR TITLE
Update Flow Core Contracts to work with Secure Cadence

### PR DESCRIPTION
**Pull request by janezpodhostnik**
Contract and transaction upgrades needed because of breaking changes in cadence when upgrading from Cadence v0.23.2 to Secure Cadence.

The go version of all packages (and the CI) was upgraded to 1.16, since in 1.15 there were build issues.

⚠️ Do not merge

These changes are needed for Secure Cadence.

Other related PRs/tags:
- cadence: [secure-cadence-m5](https://github.com/onflow/cadence/releases/tag/secure-cadence-m5)
- flow-go: [Secure Cadence](https://github.com/onflow/flow-go/pull/2337) 
- emulator: [Upgrade emulator to use Secure Cadence](https://github.com/onflow/flow-emulator/pull/127) 
- flow-go-sdk: [Secure cadence](https://github.com/onflow/flow-go-sdk/pull/278)
- flow-core-contracts: [Update cadence version to secure-cadence-m4](https://github.com/onflow/flow-core-contracts/pull/287)

### CODE REVIEW

File: lib/go/test/flow_lockedtokens_test.go [hunk 1]
```diff
@@ -2208,10 +2208,10 @@ func TestLockedTokensUnlockMultipleAccounts(t *testing.T) {
 			[][]byte{jsoncdc.MustEncode(cadence.Address(adminAddress))},
 		).(cadence.Dictionary).Pairs
 		assertEqual(t, 2, len(result))
-		assertEqual(t, nonLockedAddress.String(), result[1].Key.String()[2:])
-		assertEqual(t, CadenceUFix64("20000.0"), result[1].Value)
-		assertEqual(t, nonLocked2Address.String(), result[0].Key.String()[2:])
-		assertEqual(t, CadenceUFix64("30000.0"), result[0].Value)
+		assertEqual(t, nonLockedAddress.String(), result[0].Key.String()[2:])
```
Comment (janezpodhostnik) [inline lib/go/test/flow_lockedtokens_test.go:2211]: not sure why this happened, but the entries are now in reverse order.

**Review by joshuahannan — APPROVED** [review]
Added some changes @janezpodhostnik . It looks good to me!

File: lib/go/test/flow_lockedtokens_test.go (hunk 1 above)
Comment (janezpodhostnik) [reply lib/go/test/flow_lockedtokens_test.go:2211]: I know now that this is expected due to the change in how dictionaries are stored.

**Comment by joshuahannan** [comment]
@janezpodhostnik Is this ready to merge, or are we going to wait until the spork?

**Comment by janezpodhostnik** [comment]
Merging this now, since secure cadence is now on flow-go master

**Pull request merged**